### DISCUSSION TITLE
Pass custom encoding to loadNodeContent method

### DISCUSSION THREAD

**Discussion by alexandrtovmach** [gatsbyjs/gatsby#32738] [Ideas / Feature Requests]
## Summary

Pass custom encoding as argument to `loadNodeContent` method. Currently, it's hardcoded as `utf-8`, and there is no way to load different content.

### Basic example
I'm using `loadNodeContent` to get content from files:
![image](https://user-images.githubusercontent.com/28801003/89067432-eb45a380-d377-11ea-9084-00479f073ccc.png)

Files used `windows-1251` encoding:
![image](https://user-images.githubusercontent.com/28801003/89067628-58f1cf80-d378-11ea-89d2-da5d99e16a13.png)

After loading with `loadNodeContent` symbols become broken:
![image](https://user-images.githubusercontent.com/28801003/89067445-f698cf00-d377-11ea-9890-bfd981f4a9fd.png)

### Motivation

I faced with project that used `windows-1251` encoding in files, and I believe that's helpful for other Japanese/Chinese symbol encodings. Also it's useful for _transformer_ plugins development.

I'll prepare PR with this feature. There is just few simple changes in [`gatsby-source-filesystem`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/src/index.js#L3-L5) and [`gatsby core`](https://github.com/gatsbyjs/gatsby/blob/d842f8c27c0c65d3c6d6a1113bfe12596f93c13c/packages/gatsby/src/db/nodes.ts#L7)

P.S. This project from 2000s, so sorry for deprecated HTML markup on screenshots :man_shrugging:

**Comment by vladar** [comment]
Thanks for opening this, @alexandrtovmach !

This sounds like a valid problem. But I am not sure if adding encoding as an argument to `loadNodeContent` is the best way to go here. `loadNodeContent` is basically an interface that encapsulates knowledge on how to load any node (including binary streams).

So I think the information about an encoding should be somehow embedded in the node itself. Probably the same way we do with mimeTypes. Then the plugin that is responsible for actually loading the node could properly convert it to `utf-8`.

I realize that the scope for this change would be much bigger and not 100% sure I am right here, so let's wait for others input, especially @pieh

**Comment by pieh** [comment]
> So I think the information about an encoding should be somehow embedded in the node itself. Probably the same way we do with mimeTypes. Then the plugin that is responsible for actually loading the node could properly convert it to `utf-8`.

I'm not sure if this would work for filesystem to set encoding on its own, it would need to be hinted by users (or plugins calling `loadNodeContent`) in some way what encoding to use. Also this would require breaking change as right now we return decoded string and not pre-decoding buffer.

Given above, I would be ok with optional 2nd param (as done in https://github.com/gatsbyjs/gatsby/pull/26174) to force non-utf-8 decoding controlled by transformer plugin (I think most transformers that would use it would be pretty custom, because of how unwieldy it would be to make this user configurable via plugin options and use cases that needs this - most of data is utf-8 these days)
